app/src/main/java/com/github/se/icebreakrr/ui/authentication/ProfileCreation.kt

```diff
@@ -0,0 +1,314 @@
+                  OutlinedButton(
+                      onClick = { selectedGender = gender },
+                      modifier = Modifier.weight(1f).padding(horizontal = 4.dp),
+                      colors =
+                          ButtonDefaults.outlinedButtonColors(
+                              containerColor =
+                                  if (selectedGender == gender)
+                                      MaterialTheme.colorScheme.primaryContainer
+                                  else MaterialTheme.colorScheme.surface)) {
+                        Text(
+                            text = gender.displayName,
+                            style = MaterialTheme.typography.bodyMedium,
+                            color =
+                                if (selectedGender == gender)
+                                    MaterialTheme.colorScheme.onPrimaryContainer
+                                else MaterialTheme.colorScheme.onSurface)
+                      }
+                }
+              }
+
+          TagSelector(
+              selectedTag = selectedTags.map { tag -> Pair(tag, tagsViewModel.tagToColor(tag)) },
+              outputTag =
+                  tagsSuggestions.value.map { tag -> Pair(tag, tagsViewModel.tagToColor(tag)) },
+              stringQuery = stringQuery.value,
+              expanded = expanded,
+              onTagClick = { tag ->
+                tagsViewModel.removeFilter(tag)
+                isModified = true
+              },
+              onStringChanged = {
+                tagsViewModel.setQuery(it, selectedTags)
+                isModified = true
+              },
+              textColor = MaterialTheme.colorScheme.onSurface,
+              onDropDownItemClicked = { tag ->
+                tagsViewModel.addFilter(tag)
+                isModified = true
+              },
+              height = screenHeight,
+              width = screenWidth,
+              textSize = (screenHeight.value * 0.03).sp)
+        }
+
+    // Add confirmation button in top right corner
+    IconButton(
+        onClick = {
+          if (isFormValid()) {
+            val currentUser = Firebase.auth.currentUser
+            currentUser?.let { user ->
+              FirebaseMessaging.getInstance().token.addOnCompleteListener { task ->
+                if (task.isSuccessful) {
+                  val fcmToken = task.result
+                  val newProfile =
+                      Profile(
+                          uid = user.uid,
+                          name = fullName.value,
+                          gender = selectedGender!!, // Add this line
+                          birthDate =
+                              Timestamp(
+                                  Calendar.getInstance()
+                                      .apply {
+                                        time = selectedDate.value!!
+                                        set(Calendar.HOUR_OF_DAY, 0)
```

Comment:
I think you have already set hour/minute/seconds/milliseconds on the pickedDate variable, so no need to set it again.

Reply:
yep!